### PR TITLE
refactor get_asset_check_executions

### PR DESCRIPTION
**Pull request by johannkm**
Rename and take an AssetCheckKey

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16930** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16930" target="_blank">Graphite</a> 
    * **PR #16931** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16931" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16931?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/storage/legacy_storage.py [hunk 1]
```diff
@@ -607,16 +608,14 @@ def free_concurrency_slots_for_run(self, run_id: str) -> None:
     def free_concurrency_slot_for_step(self, run_id: str, step_key: str) -> None:
         return self._storage.event_log_storage.free_concurrency_slot_for_step(run_id, step_key)
 
-    def get_asset_check_executions(
+    def get_asset_check_execution_history(
         self,
-        asset_key: AssetKey,
-        check_name: str,
+        key: "AssetCheckKey",
```
Comment (rexledesma) [inline python_modules/dagster/dagster/_core/storage/legacy_storage.py:613]: `key` is very ambiguous. We cannot expect the method name to hint at what sort of key we want here.

```suggestion
        asset_check_key: "AssetCheckKey",
```

**Review by rexledesma — COMMENTED** [review]
Looks fine, let's just clarify the parameter ambiguity. Will approve after

**Review by rexledesma — APPROVED** [review]
We should standardize on `asset_check_key` or `check_key` in the codebase. Both currently appear. I believe you can unify this in a later diff. 

Will approve to unblock

**Pull request merged**